File: robots/csharp/Robot.cs
namespace ReportGeneratorCS
{
    public class Robot
    {
        private string bin;
        private Machine location;

        public Machine Location()
        {
            return location;
        }

        public void MoveTo(Machine location)
        {
            this.location = location;
        }

        public void Pick()
        {
            bin = location.Take();
        }

        public string Bin()
        {
            return bin;
        }

        public void Release()
        {
            location.Put(bin);
            bin = null;
        }
    }
}

File: robots/javascript/robot.js
function Robot() {
	this.location = null;
	this.bin = null;
}

Robot.prototype = {
  move_to: function(location){
    this.location = location;
  },

  pick: function() {
    this.bin = this.location.take();
  },

  release: function() {
    this.location.put(this.bin);
    this.bin = null;
  },

  toString: function() {
  	return '[object Robot]';
  }
}